<?xml version="1.0"?>
<!-- vim: set fenc=utf-8 et sts=4 ft=xml : -->
<launch>

    <!-- begin test arguments -->
    <machine name="search"
             address="localhost"/>
    <machine name="navigate"
             address="localhost"/>
    <!-- end test arguments -->
    
    <include file="$(find executive)/launch/executive.launch"/>

    <!-- crap below here is for test only -->

    <node
        name="navigate_audio"
        pkg="sound_play"
        type="soundplay_node.py"
        ns="/io/audio">
        <remap from="robotsound" to="navigate_soundplay"/>
    </node>
    
    <include file="$(find samplereturn)/launch/joystick.launch"/>

    <include file="$(find samplereturn)/launch/planner.launch">
        <arg name="map_config" value="maps/map_park_fake.yaml"/>
    </include>
        
    <node
        name="navigate_announcer"
        pkg="samplereturn"
        type="announcer.py"
        ns="/io/audio">
        <remap from="audio_in" to="navigate"/>
        <remap from="audio_out" to="navigate_soundplay"/>
    </node>

    <!-- joint state publishing -->
    <param name="robot_description"
           command="$(find xacro)/xacro.py '$(find samplereturn)/urdf/robot.xacro'" />

    <node name="state_publisher"
          pkg="robot_state_publisher" 
          type="robot_state_publisher"
          output="log"
          ns="/motion">
        <remap from="joint_states" to="platform_joint_state"/>
        <param name="publish_frequency" type="double" value="10"/>

    </node>
    
</launch>
